lib/radiator/directory/editor.ex

```diff
@@ -736,4 +748,88 @@ defmodule Radiator.Directory.Editor do
       @not_authorized_match
     end
   end
+
+  def list_contribution_roles() do
+    {:ok, Repo.all(Radiator.Contribution.Role)}
+  end
+
+  def list_contributions(actor, _subject = %Podcast{id: id}) do
+    with {:ok, subject} <- get_podcast(actor, id) do
+      {:ok, preloaded_contributions(Ecto.assoc(subject, :contributions))}
+    end
+  end
+
+  def list_contributions(actor, _subject = %Audio{id: id}) do
+    with {:ok, subject} <- get_audio(actor, id) do
+      {:ok, preloaded_contributions(Ecto.assoc(subject, :contributions))}
+    end
+  end
+
+  defp preloaded_contributions(query) do
+    from(c in query,
+      preload: [:person, :role]
+    )
+    |> Repo.all()
+    |> Enum.sort(fn a, b ->
+      cond do
+        a.role_id == b.role_id ->
+          a.position < b.position
+
+        true ->
+          a.role_id < b.role_id
+      end
+    end)
+  end
+
+  def create_contribution(actor, subject, attrs) do
+    with :ok <- with_permission(actor, subject, :edit) do
+      Editor.Editor.create_contribution(subject, attrs)
+    end
+  end
+
+  def get_contribution(actor, id) do
+    with {:ok, contribution, subject} <- get_contribution_and_subject(actor, id),
+         :ok <- with_permission(actor, subject, :readonly) do
+      {:ok, contribution}
+    end
+  end
+
+  def delete_contribution(actor, id) do
+    with {:ok, contribution, subject} <- get_contribution_and_subject(actor, id),
+         :ok <- with_permission(actor, subject, :edit) do
+      Repo.delete(contribution)
+    end
+  end
+
+  defp get_contribution_and_subject(actor, id) do
+    case Repo.get(AudioContribution, id) || Repo.get(PodcastContribution, id) do
```

Comment:
Does this not have an edge case where both ids are identical, making both conditions true, making it impossible to get the PodcastContribution because the AudioContribution is always returned first?

Reply:
Ah, saw the migration now, which ensures unique IDs over both tables.

Reply:
Yeah - went through the same process ^^